src/Handler/LogDnaHandler.php

```diff
@@ -68,6 +70,28 @@ protected function getDefaultFormatter(): FormatterInterface
 
     public function write(array $record)
     {
+        $body = $record['formatted'];
+
+        if (mb_strlen($body) > static::LOGDNA_BYTE_LIMIT) {
+            $decodedBody = json_decode($body, true);
+
+            $body = json_encode([
+                'lines' => [
+                    [
+                        'timestamp' => $decodedBody['lines'][0]['timestamp'] ?? '',
+                        'line'      => $decodedBody['lines'][0]['line'] ?? '',
+                        'app'       => $decodedBody['lines'][0]['app'] ?? '',
+                        'level'     => $decodedBody['lines'][0]['level'] ?? '',
+                        'meta'      => [
+                            'longException' => mb_substr($body, 0, static::LOGDNA_BYTE_LIMIT),
```

Comment:
Is it a byte limit or a multibyte character limit? Does that impact what we should do here?

Reply:
@WillJW I think this function call is correct, because we want to slice based on the fact there may be multibyte characters in the string. However, the `mb_strlen()` call above should be `strlen()`, because we want the number of bytes, **not** the number of multibyte characters.

Reply:
@WillJW i'm not 100% sure and can't see anything documented, it looks like `mb_substr` treats unicode chars as 1 char, do you think we should use `strlen` and `substr` here instead? I imagine this should all be mostly ascii anyway and i've left 2000 bytes of leeway so maybe its not a huge issue

Reply:
@liamkeily I'm not 100% sure either way. @ziadoz sounds like he's got it sussed, so I'd go with his suggestion.

Reply:
@ziadoz good points, actually i think if it just use `8bit` encoding on the mb_ functions it will do thae job

Reply:
Changed this to use '8bit' encoding